components/SearchResult.tsx

```diff
@@ -1,18 +1,16 @@
 import { useQuery } from "@tanstack/react-query";
-import { SearchResponseSchema } from "@/app/utils/discogs.schemas";
 import Result from "./Result";
+import { Album } from "@/app/utils/discogs.schemas";
 
 export default function SearchResult({ searchValue }: { searchValue: string }) {
   const { data, isError, isLoading } = useQuery({
     queryKey: ["search", searchValue],
     queryFn: async () => {
       const params = new URLSearchParams({ search: searchValue });
-      const result = await fetch("/api/search?" + params).then((res) =>
-        res.json()
-      );
+      const res = await fetch("/api/search?" + params);
+      const data = await res.json();
 
-      const parsedResult = SearchResponseSchema.parse(result);
```

Comment:
you still need to parse this using `SearchResponseSchema`

Reply:
This is to ensure that there is a check at runtime right?

Reply:
yeah it's a runtime check, and it actually gives type data (since fetch returns any)